hydromt_wflow/wflow.py

```diff
@@ -2494,6 +2498,292 @@ def setup_rootzoneclim(
+        population_method: str = "sum",
+    ):
+        """_summary_.
+
+        _extended_summary_
+
+        Parameters
+        ----------
+        non_irigation_fn : str, optional
+            _description_, by default "pcr_globwb"
+        non_irigation_vars : list, optional
+            _description_, by default ["dom", "ind", "lsk"]
+        non_irigation_year : int, optional
+            _description_, by default None
+        non_irigation_method : str, optional
+            _description_, by default "nearest"
+        population_fn : str, optional
+            _description_, by default "worldpop_2020_constrained"
+        population_method : str, optional
+            _description_, by default "sum"
+
+        Raises
+        ------
+        ValueError
+            _description_
+        """
+        self.logger.info("Preparing non irigation demand maps.")
+        if not all([item in ["dom", "ind", "lsk"] for item in non_irigation_vars]):
+            raise ValueError("")
+
+        # Set flag for cyclic data
+        _cyclic = False
+
+        # Selecting data
+        if non_irigation_year is None:
+            non_irigation_year = 2005
+        non_irigation_raw = self.data_catalog.get_rasterdataset(
+            non_irigation_fn,
+            geom=self.region,
+            buffer=2,
+            variables=[
+                f"{var}_{mode}"
+                for var, mode in product(non_irigation_vars, ["gross", "net"])
+            ],
+            version=non_irigation_year,
+        )
+        if "time" in non_irigation_raw.coords:
+            _cyclic = True
+            non_irigation_raw["time"] = non_irigation_raw.time.astype("int32")
+
+        if _cyclic and self.get_config("input.cyclic") is None:
+            self.set_config("input.cyclic", [])
+
+        # Get population data
+        pop_raw = self.data_catalog.get_rasterdataset(
+            population_fn,
+            geom=self.region,
+            buffer=2,
+        ).raster.mask_nodata()
+
+        # Create static water demand rasters
+        non_irigation, popu = workflows.demand.non_irigation(
+            non_irigation_raw,
+            ds_like=self.grid,
+            ds_method=non_irigation_method,
+            popu=pop_raw,
+            popu_method=population_method,
+        )
+        self.set_grid(non_irigation)
+        self.set_grid(popu)
+
+        # Update the settings toml with non irigation stuff
+        for var in non_irigation.data_vars:
+            sname, suffix = var.split("_")
+            lname = workflows.demand.map_vars[sname]
+            self.set_config(
+                f"input.vertical.{lname}.demand_{suffix}",
+                var,
+            )
+
+            # Also for the fact that these parameters are cyclic
+            if _cyclic:
+                self.config["input"]["cyclic"].append(
+                    f"input.vertical.{lname}.demand_{suffix}",
+                )
+
+    def setup_irrigation(
```

Comment:
I did not review this function and workflows yet, not sure if it is finished.
One thing while having a quick look, is about the consistency of all the datasets irrigated_area, landuse and mirca together with the already existing landuse map of wflow and other landuse paramters (setup_lulcmaps)?
I didn't check anything with mirca. but for irrigated and paddy areas, I guess good to burn in the wflow_landuse map and re-derive parameters accordingly?

Reply:
This is a good point, but it was fairly tricky to find a dataset that had both good global coverage of (high resolution) irrigated area's and a mask for paddy fields. Hence why we decided to combine a couple of different datasources. I guess we can add support for the glcnmo landcover dataset, as that contains quite high resolution data and a specific classification for rice fields.